modules/heat_conduction/test/tests/gap_heat_transfer_mortar/closed_gap_prescribed_pressure.i

```diff
@@ -0,0 +1,191 @@
+    block = 'interface_secondary_subdomain'
+  []
+[]
+
+[AuxVariables]
+  [interface_normal_lm]
+    order = FIRST
+    family = LAGRANGE
+    block = 'interface_secondary_subdomain'
+    initial_condition = 100.0
+  []
+[]
+
+[Kernels]
+  [HeatDiff_steel]
+    type = ADHeatConduction
+    variable = temperature
+    thermal_conductivity = steel_thermal_conductivity
+    block = 'left_block'
+  []
+  [HeatDiff_aluminum]
+    type = ADHeatConduction
+    variable = temperature
+    thermal_conductivity = aluminum_thermal_conductivity
+    block = 'right_block'
+  []
+[]
+
+[BCs]
+  [temperature_left]
+    type = ADDirichletBC
+    variable = temperature
+    value = 800
+    boundary = 'moving_block_left'
+  []
+  [temperature_right]
+    type = ADDirichletBC
+    variable = temperature
+    value = 250
+    boundary = 'fixed_block_right'
+  []
+[]
+
+[Constraints]
+  [thermal_contact]
+    type = ModularGapConductanceConstraint
+    variable = temperature_interface_lm
+    secondary_variable = temperature
+    primary_boundary = moving_block_right
+    primary_subdomain = interface_primary_subdomain
+    secondary_boundary = fixed_block_left
+    secondary_subdomain = interface_secondary_subdomain
+    gap_flux_models = 'closed'
+  []
+[]
+
+[Materials]
+  [steel_thermal_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'steel_density steel_thermal_conductivity steel_hardness'
+    prop_values = '8e3            16.2                       129' ## for stainless steel 304
+    block = 'left_block'
+  []
+
+  [aluminum_thermal_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'aluminum_density aluminum_thermal_conductivity aluminum_hardness'
+    prop_values = ' 2.7e3           210                             15' #for 99% pure Al
+    block = 'right_block'
+  []
+[]
+
+[UserObjects]
+  [closed]
+    type = GapFluxModelPressureDependentConduction
+    primary_conductivity = steel_thermal_conductivity
+    secondary_conductivity = aluminum_thermal_conductivity
+    temperature = temperature
+    normal_pressure = interface_normal_lm
+    primary_hardness = steel_hardness
+    secondary_hardness = aluminum_hardness
+    boundary = moving_block_right
+  []
+[]
+
+[Postprocessors]
+  [steel_pt_interface_temperature]
+    type = NodalVariableValue
+    nodeid = 245
+    variable = temperature
+  []
+  [aluminum_pt_interface_temperature]
+    type = NodalVariableValue
+    nodeid = 657
+    variable = temperature
+  []
+  [interface_heat_flux_steel]
+    type = ADSideDiffusiveFluxAverage
+    variable = temperature
+    boundary = moving_block_right
+    diffusivity = steel_thermal_conductivity
+  []
+  [interface_heat_flux_aluminum]
+    type = ADSideDiffusiveFluxAverage
+    variable = temperature
+    boundary = fixed_block_left
+    diffusivity = aluminum_thermal_conductivity
+  []
+[]
+
+
+[Executioner]
+  type = Steady
+  solve_type = NEWTON
+  automatic_scaling = false
+
+  nl_forced_its = 1
```

Comment:
Is this needed in this input file? If it is, it may point to a complete satisfaction of the equations via initial conditions?